openquakeplatform_ipt/static/ipt/js/app_web.js

```diff
@@ -0,0 +1,298 @@
+        else {
+            families[subdir] = [el];
+        }
+    }
+
+    function subdir_ls_cbgen(k, family_ext)
+    {
+        var family = family_ext.slice();
+
+        return function subdir_ls_cb(uuid, msg) {
+            var options = [];
+            var app_msg;
+
+            app_msg = msg.result;
+            for (var i = 0 ; i < app_msg.content.length ; i++) {
+                var v = app_msg.content[i];
+                options.push([k + '/' + v, v]);
+            }
+            for (var i = 0 ; i < family.length ; i++) {
+                select_update(family[i], options);
+            }
+        }
+    }
+
+     for (var k in families) {
+        if (families.hasOwnProperty(k)) {
+            var family_ = families[k].slice();
+            {
+                var family = family_;
+
+                subdir_ls_cb = subdir_ls_cbgen(k, family);
+            }
+            gem_api.ls(subdir_ls_cb, k);
+        }
+    }
+}
+
+var track_status_ct = 0;
+function track_status_cb(uuid, msg)
+{
+    console.log('track_status of QGIS connection (no UI at the moment): ' + (msg.success ? '' : 'NOT ') + 'CONNECTED');
+    track_status_ct++;
+
+    if (msg.success) {
+        if (typeof gem_api != 'undefined') {
+            // set folders to save collected files
+            function init_ls_cb(uuid, msg) {
+                console.log('init_ls_cb');
+                if (!msg.complete) {
+                    return;
+                }
+
+                app_msg = msg.result;
+                if (app_msg.success != true)
+                    return manage_error_cb(uuid, app_msg);
+
+                var ct_sync = allowed_dirs.length;
+
+                for (var i = 0 ; i < allowed_dirs.length ; i++) {
+                    all_dir = allowed_dirs[i];
+
+                    if (app_msg.content.indexOf(all_dir + '/') == -1) {
+                        // folder not found, create it
+                        function init_mkdir_cb(uuid, app_msg)
+                        {
+                            ct_sync--;
+                            if (ct_sync == 0) {
+                                populate_selects();
+                            }
+                        }
+                        gem_api.mkdir(init_mkdir_cb, all_dir);
+                    }
+                    else {
+                        ct_sync--;
+                        if (ct_sync == 0) {
+                            populate_selects();
+                        }
+                    }
+                }
+            }
+            gem_api.ls(init_ls_cb);
+        }
+
+        console.log("msg_close ...");
+        gem_ipt.qgis_msg_close();
+    }
+    else {
+        console.log("msg_open ...");
+        gem_ipt.qgis_msg_open('The web-application was required by a browser with an enabled OpenQuake extension but no QGIS application is currently running, please:\nLaunch QGIS - OR - disable the OpenQuake extension clicking its icon on the browser and reloading the page.');
```

Comment:
I suggest to modify the message as follows:
'The web-application was required by a browser with an enabled OpenQuake extension, but no QGIS application is currently running. Please:\nlaunch QGIS - OR - disable the OpenQuake extension by clicking its icon on the browser and reloading the page.'